test/utils/deep-partial.test.ts

```diff
@@ -17,4 +18,14 @@ describe('DeepPartial', function () {
     };
     expect(test).to.exist;
   });
+
+  it('should not allow broken deep partial definitions', async function () {
+    type TData = {
+      value: Date
+    }
+    const test: DeepPartial<TData> = {
+      value: 1 // Should not compile!
```

Comment:
@SylvainEstevez notice how this is currently legal 😱

Reply:
Notice how it's now properly detected to be broken: https://travis-ci.org/github/bluebirds-blue-jay/utils/jobs/730652271

Reply:
Wow, I love the new error descriptions!